cmd/exit_broadcast.go

```diff
@@ -52,15 +56,44 @@ func newBcastFullExitCmd(runFunc func(context.Context, exitConfig) error) *cobra
 		{lockFilePath, false},
 		{validatorKeysDir, false},
 		{exitEpoch, false},
-		{validatorPubkey, true},
+		{validatorPubkey, false},
 		{beaconNodeEndpoints, true},
 		{exitFromFile, false},
+		{exitFromDir, false},
 		{beaconNodeTimeout, false},
 		{publishTimeout, false},
 	})
 
 	bindLogFlags(cmd.Flags(), &config.Log)
 
+	wrapPreRunE(cmd, func(cmd *cobra.Command, _ []string) error {
+		valPubkPresent := cmd.Flags().Lookup(validatorPubkey.String()).Changed
+		exitFilePresent := cmd.Flags().Lookup(exitFromFile.String()).Changed
+		exitDirPresent := cmd.Flags().Lookup(exitFromDir.String()).Changed
+
+		if !valPubkPresent && !config.All {
+			//nolint:revive,perfsprint // we use our own version of the errors package; keep consistency with other checks.
+			return errors.New(fmt.Sprintf("%s must be specified when exiting single validator.", validatorPubkey.String()))
```

Comment:
I think you should be using `errors.New` parameters instead of `Sprintf`, e.g.

```
return errors.New("validator pubkey must be specified when exiting single validator", z.Str("pubkey", validatorPubkey.String()))
```

Check other instances below.

Reply:
Hmmm I was using what we had for the already existing exit messages ([like here](https://github.com/ObolNetwork/charon/blob/8364cf20505a14a5715af6f225dc1203e7c98a6c/cmd/exit_sign.go#L67)). Here it's not exactly parameters as we do in other error messages, as this is a constant and somehow part of the message. I'd rather add its value here (in this case `"validator-public-key"`) than putting it as a parameter.